### PR TITLE
fix: neon adoon support kb 0.9

### PR DESCRIPTION
**Pull request by skyrise-l**

### CODE REVIEW

File: addons/neon/templates/clusterdefinition.yaml [hunk 1]
```diff
@@ -76,16 +101,15 @@ spec:
...
                 /compute.sh
-
-    - name: neon-storagebroker
+    - name: neon-storbroker
       workloadType: Stateful
       characterType: storagebroker
       podSpec:
         containers:
-          - name: neon-storagebroker
+          - name: neon-storbroker
```
Comment (Y-Rookie) [inline addons/neon/templates/clusterdefinition.yaml:109]: for bacward compatible,  the container name should not be renamed

File: addons/neon/templates/clusterdefinition.yaml [hunk 2]
```diff
@@ -95,7 +119,7 @@ spec:
       characterType: safekeeper
       componentDefRef:
         - &storagebrokerRef
-          componentDefName: neon-storagebroker
+          componentDefName: neon-storbroker
```
Comment (Y-Rookie) [inline addons/neon/templates/clusterdefinition.yaml:122]: componentDefName here is the name of clusterDefinition.spec.componentDefs[x].name，not the name of ComponentDefinition

File: addons/neon/templates/componentdefinition-neon-compute.yaml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+          name: postgresql
+        command: [ "/scripts/compute.sh" ]
+        volumeMounts:
+        - mountPath: /config
+          name: neon-config
+        - mountPath: /data
+          name: data
+        - name: scripts
+          mountPath: /scripts
+  serviceKind: postgresql
```
Comment (Y-Rookie) [inline addons/neon/templates/componentdefinition-neon-compute.yaml:100]: use neon-compute would be better

File: addons/neon/templates/componentdefinition-neon-pageserver.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    - command: ["/scripts/pageserver_start.sh"]
+      image: {{ include "neon-pageserver.image" . }}
+      name: neon-pageserver
+      volumeMounts:
+      - mountPath: /usr/local/neon/pageserver
+        name: neon-pageserver
+      - name: scripts
+        mountPath: /scripts
+  serviceKind: neon-pageserver
+  serviceVersion: {{ .Values.componentServiceVersion }}
```
Comment (Y-Rookie) [inline addons/neon/templates/componentdefinition-neon-pageserver.yaml:41]: is all the ComponentDefinitions has the same serviceVersion?

File: addons/neon/templates/componentdefinition-neon-pageserver.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  services:
+  - name: neon-pageserver
+    serviceName: neon-pageserver
+    spec:
+      ports:
+      - name: pageserver
```
Comment (Y-Rookie) [inline addons/neon/templates/componentdefinition-neon-pageserver.yaml:47]: defined the namePorts in runtime, and reference by port name in services

File: addons/neon/templates/componentdefinition-neon-safekeeper.yaml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+      - mountPath: /usr/local/neon/safekeeper
+        name: neon-safekeeper
+  serviceKind: neon-safekeeper
+  serviceVersion: {{ .Values.componentServiceVersion }}
+  services:
+  - name: neon-safekeeper
+    serviceName: neon-safekeeper
+    spec:
+      ports:
+      - name: safekeeper
```
Comment (Y-Rookie) [inline addons/neon/templates/componentdefinition-neon-safekeeper.yaml:52]: ditto

File: addons/neon/templates/componentdefinition-neon-pageserver.yaml (hunk 4 above)
Comment (skyrise-l) [reply addons/neon/templates/componentdefinition-neon-pageserver.yaml:41]: They all use the same image, so the serviceVersion should be consistent?

**Review by nayutah — APPROVED** [review]

**Pull request merged**